File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int> pq;
        for(auto &x : nums) pq.push(x);
        long long int ans = 0;
        while(k--){
            int mxElement = pq.top(); pq.pop();
            ans += mxElement;
            pq.push(ceil(mxElement/3.0));
        }
        return ans;
    }
};

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int x : nums) pq.add(x);
        long ans = 0;
        while(k-- > 0){
            int mxEle = pq.poll();
            ans += mxEle;
            pq.add((int)Math.ceil(mxEle/3.0));
        }
        return ans;
    }
}